src/Enigma2.cpp

```diff
@@ -295,7 +295,15 @@ PVR_ERROR Enigma2::GetEPGForChannel(ADDON_HANDLE handle, const PVR_CHANNEL &chan
   if (m_epg.IsInitialEpgCompleted() && m_settings.GetEPGDelayPerChannelDelay() != 0)
     Sleep(m_settings.GetEPGDelayPerChannelDelay());
 
-  return m_epg.GetEPGForChannel(handle, channel, iStart, iEnd);
+  if (!m_channels.IsValid(channel.iUniqueId))
+  {
+    Logger::Log(LEVEL_ERROR, "%s Could not fetch channel object - not fetching EPG for channel with UniqueID '%d'", __FUNCTION__, channel.iUniqueId);
+    return PVR_ERROR_SERVER_ERROR;
+  }
+
+  std::shared_ptr<Channel> myChannel = m_channels.GetChannel(channel.iUniqueId);
+
+  return m_epg.GetEPGForChannel(handle, myChannel, iStart, iEnd);
```

Comment:
Why not making one line out of the two above? For what do you need myChannel in a var?